addons/account_edi/models/account_edi_format.py

```diff
@@ -19,51 +19,153 @@ class AccountEdiFormat(models.Model):
 
     name = fields.Char()
     code = fields.Char()
-    hide_on_journal = fields.Selection([('import_export', 'Import/Export'), ('import', 'Import Only')], default='import_export', help='used to hide this EDI format on journals')
+    has_web_service = fields.Boolean()
 
     _sql_constraints = [
         ('unique_code', 'unique (code)', 'This code already exists')
     ]
 
+
+    ####################################################
+    # Low-level methods
+    ####################################################
+
+    @api.model
+    def create(self, vals):
+        res = super().create(vals)
+
+        journals = self.env['account.journal'].search([])
```

Comment:
Not really, we allow edi_formats to be set on any journal.

- For example, if we want to be able to send a payment without a reconciled invoice (we thought it was possible for Mexico for a short time)
- Another example would be if want to make the OCR an EDI.

We thought it would add more flexibility to allow edi on any journal, but we implemented a hook ```_edi_compatible_for_journal``` so that each edi can specify if it should appear on that journal.